### ISSUE TITLE
Querystring widget breaks edit page of Collection if page is refreshed

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#1020]
How to replicate:

- Go to https://volto.kitconcept.com/add?type=Collection
- Add a new collection, set a simple criteria with Types and Document as values
- Save the collection
- Click Edit on the collection
- Force refresh of page

What happens is: componentDidMount() calls the getQuerystring action in the browser, this populates the this.props.indexes property. But the server doesn't have the ``indexes`` populated, so code lower down in the selection filters all fail. 

My fix was to add this check at the top of the render() method:
```
    if (Object.keys(indexes).length === 0)
      return <div>Loading...</div>;
```

**Comment by sneridagh** [comment]
I can reproduce it here too. Needs to be fixed.

**Comment by sneridagh** [comment]
Fixed by #1123

**Comment by sneridagh** [comment]
@tiberiuichim I finally decided to fix it like we did in the listing block querystringwidget:
https://github.com/plone/volto/pull/1123/files#diff-a9fdf615b4534af393a47b37786f12f9R345

It wasn't only broken on SSR, but also in normal operation.
